### ISSUE TITLE
Ripgrep process causing CPU slow downs because of `external` folder if host cannot access files crated by dev container

### ISSUE THREAD

**Issue by aminya** [hedronvision/bazel-compile-commands-extractor#144]
Mirrored from https://github.com/microsoft/vscode-remote-release/issues/9110

Type: <b>Bug</b>

I have a repository with `.devcontainer.json` at the root. When I open the repository (without the dev container running). The Ripgrep process uses 100% of the CPU consistently. 

It seems it is searching for the devcontainer files. I found the issue by doing an extension bisect. 

It is the `external` folder added by Bazel-compiler-command-extractor. Inside that folder, there are huge number of files that do not exist (invalid symlink files). Because those folders are created inside the dev container, the host doesn't have access to them.

I ran the rg command that VsCode uses
```bash
 /usr/share/code-insiders/resources/app/node_modules.asar.unpacked/@vscode/ripgrep/bin/rg --hidden -case-sensitive --no-require-git -g "{.devcontainer.json,.devcontainer/devcontainer.json/.devcontainer/*/devcontainer.json}" --no-ignore --follow --no-config --no-ignore-global
```
```
./external/test/bazel-test/external/boost/bazel-bin/test.runfiles/test.txt: No such file or directory (os error 2)
# the error is repeated many times
```

That's why adding `external` and `bazel-*` fixes the issue.

```json
  "files.exclude": {
    "**/.git": true,
    "**/.cache": true,
    ".idea/": true,
    "external/": true,
    "bazel-*/": true,
    "**/build": true,
  }
```

Bazel removes/adds these runtime files depending on the circumstances. It's a very slow behaviour, but unfortunately, I cannot change that.

Although these folders are gitignored, `ripgrep` still searches for those due to the flags `VsCode` is passing

![image](https://github.com/microsoft/vscode-remote-release/assets/16418197/a06df996-63f6-486b-bb73-eada3e5cec7d)

Here is `.devcontainer.json` in case it helps (although I didn't actually open the dev container!)
```json
{
    "name": "Dev Container",
    "dockerComposeFile": "${localWorkspaceFolder}/docker-compose.yaml",
    "service": "app",
    "workspaceFolder": "/home/aminya/",
    "remoteUser": "aminya",
    "features": {
        "ghcr.io/devcontainers/features/common-utils:2": {
            "configureZshAsDefaultShell": true
        }
    },
    "customizations": {
        "vscode": {
            "extensions": [
				"BazelBuild.vscode-bazel",
				"janisdd.vscode-edit-csv",
				"llvm-vs-code-extensions.vscode-clangd",
				"ms-vscode-remote.remote-containers",
				"ms-vscode.cmake-tools",
				"ms-vscode.cpptools",
				"pustelto.bracketeer",
				"smilerobotics.urdf",
				"stackbuild.bazel-stack-vscode",
				"twxs.cmake",
				"vadimcn.vscode-lldb",
				"josetr.cmake-language-support-vscode",
				"cheshirekow.cmake-format",
				"GitHub.copilot",
				"GitHub.copilot-chat",
				"GitHub.copilot-labs"
			]
        }
    }
}

```

Extension version: 0.318.0
VS Code version: Code - Insiders 1.84.0-insider (4678919d52909efc7c4ab958a2e5c8f45d50ffc3, 2023-10-13T05:33:39.403Z)
OS version: Linux x64 6.2.0-34-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz (16 x 3599)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|6, 8, 10|
|Memory (System)|30.97GB (25.74GB free)|
|Process Argv|/home/aminya/Gitlab/robodrake|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details>

I verified that the issue happens on stable VsCode as well.

**Comment by cpsauer** [comment]
Hey Amin--sorry I've been so slow on this. I've had some urgent things I've needed to do, so I had to step away for a little bit.

Any thoughts about what we should or could do about this? We're going to need that external folder; I think we're going to need to ask ripgrep to be robust to this. Seems like perhaps a part of the problem is that ripgrep isn't respecting the gitignore, as in the other thread. Could we get the ripgrep extension to fix that?

(If that still doesn't work, we might also want to check that ripgrep is respecting reading from the hidden gitignore, $(git rev-parse --git-common-dir)/info/exclude, since that's a place where we (and other tools) ought to write things.
